keps/sig-cluster-lifecycle/generic/1755-communicating-a-local-registry/README.md

```diff
@@ -0,0 +1,545 @@
+
+This KEP proposes a standard for how these clusters should expose their support
+for this feature, so that tooling can interoperate with them without redundant
+configuration.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals and non-goals of
+this KEP.  Describe why the change is important and the benefits to users.  The
+motivation section can optionally provide links to [experience reports][] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+Many local clusters currently support local registries. But they put the onus of
+configuration on the user. First, the user has to follow instructions to setup
+the registry in the cluster.
+
+- https://kind.sigs.k8s.io/docs/user/local-registry/
+- https://microk8s.io/docs/registry-built-in
+- https://github.com/rancher/k3d/blob/master/docs/registries.md#using-a-local-registry
+- https://minikube.sigs.k8s.io/docs/handbook/registry/#enabling-insecure-registries
+
+When the cluster has been set up, the user has to manually go through each
+individual tool that needs to push images to the cluster, and configure it with
+the hostname and port of the new registry.
+
+The motivation of this KEP is to remove this configuration redundancy, and
+reduce the burden on the user to configure all their image tools.
+
+### Goals
+
+<!--
+List the specific goals of the KEP.  What is it trying to achieve?  How will we
+know that this has succeeded?
+-->
+
+- Agree on a standard way for cluster configuration tools to record how
+  developer tools should interact with the local registry.
+  
+- Agree on a standard way for developer tools to read that information
+  when pushing images to the cluster.
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP?  Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+- Modifying how local registries are currently implemented on these clusters.
+
+- How this might be extended to support multiple registries. We assume clusters
+  have at most one registry, because that's how all existing implementations
+  work.
+
+- Any API for configuring a local registry in a cluster.
+  If there was a standard CRD that configured a local registry, and we could
+  convince all implementations to support that CRD, this KEP would become moot.
+  That approach would have substantial technical challenges.
+  
+- A general-purpose mechanism for publicly exposing cluster configuration.
+
+- A general-purpose mechanism for cluster capability detection for developer
+  tooling. The proposer of this KEP maintains a fair amount of code in Tilt that
+  does cap detection on clusters. I have a lot of interest in proposing such a
+  mechanism!  But I also think that best practices in this space are moving
+  fast. Local registries are the only best practice that's mature enough where
+  this makes sense.
```

Comment:
potentially the following would suffice for the last point?
> A general-purpose mechanism for cluster capability detection for developer
  tooling. With the best practices around cluster capability detection moving fast, local registries are the only best practice that's mature enough where this makes sense.

ideally KEPs should not have personalizations e.g. "I do".